test/groovy/ArtifactSetVersionTest.groovy

```diff
@@ -0,0 +1,115 @@
+        jscr.setReturnValue("date +'%Y%m%d%H%M%S'", '20180101010203')
+
+        cpe = loadScript('commonPipelineEnvironment.groovy').commonPipelineEnvironment
+        artifactSetVersionScript = loadScript("artifactSetVersion.groovy")
+
+        gitUtils = new GitUtils()
+        prepareObjectInterceptors(gitUtils)
+    }
+
+    @Test
+    void testVersioning() {
+        artifactSetVersionScript.call(script: [commonPipelineEnvironment: cpe], juStabGitUtils: gitUtils, buildTool: 'maven', gitSshUrl: 'myGitSshUrl')
+
+        assertEquals('1.2.3-20180101010203_testCommitId', cpe.getArtifactVersion())
+        assertEquals('testCommitId', cpe.getGitCommitId())
+
+        assertEquals('mvn versions:set -DnewVersion=1.2.3-20180101010203_testCommitId --file pom.xml', jscr.shell[2])
+        assertEquals('git add .', jscr.shell[3])
+        assertEquals ("git commit -m 'update version 1.2.3-20180101010203_testCommitId'", jscr.shell[4])
+        assertEquals ("git remote set-url origin myGitSshUrl", jscr.shell[5])
+        assertEquals ("git tag build_1.2.3-20180101010203_testCommitId", jscr.shell[6])
+        assertEquals ("git push origin build_1.2.3-20180101010203_testCommitId", jscr.shell[7])
+    }
+
+    @Test
+    void testVersioningCustomGitUserAndEMail() {
+        artifactSetVersionScript.call(script: [commonPipelineEnvironment: cpe], juStabGitUtils: gitUtils, buildTool: 'maven', gitSshUrl: 'myGitSshUrl', gitUserEMail: 'test@test.com', gitUserName: 'test')
+
+        assertEquals ('git config user.name "test"', jscr.shell[4])
+        assertEquals ('git config user.email "test@test.com"', jscr.shell[5])
+    }
+
+    @Test
+    void testVersioningWithTimestamp() {
+        artifactSetVersionScript.call(script: [commonPipelineEnvironment: cpe], juStabGitUtils: gitUtils, buildTool: 'maven', timestamp: '2018')
+        assertEquals('1.2.3-2018_testCommitId', cpe.getArtifactVersion())
+    }
+
+    @Test
+    void testVersioningNoBuildTool() {
+        thrown.expect(Exception)
+        thrown.expectMessage('ERROR - NO VALUE AVAILABLE FOR buildTool')
+        artifactSetVersionScript.call(script: [commonPipelineEnvironment: cpe], juStabGitUtils: gitUtils)
+    }
+
+    @Test
+    void testVersioningWithCustomTemplate() {
+        artifactSetVersionScript.call(script: [commonPipelineEnvironment: cpe], juStabGitUtils: gitUtils, buildTool: 'maven', versioningTemplate: '${version}-xyz')
+        assertEquals('1.2.3-xyz', cpe.getArtifactVersion())
+    }
+
+    @Test
+    void testVersioningWithTypeAppContainer() {
+        cpe.setArtifactVersion('1.2.3-xyz')
+        artifactSetVersionScript.call(script: [commonPipelineEnvironment: cpe], juStabGitUtils: gitUtils, buildTool: 'docker', artifactType: 'appContainer', dockerVersionSource: 'appVersion')
+        assertEquals('1.2.3-xyz', cpe.getArtifactVersion())
+        assertEquals('1.2.3-xyz', jwfr.files['VERSION'])
+    }
+
+    void prepareObjectInterceptors(object) {
```

Comment:
This method is duplicated in serveral test classes. What about a TestUtil class?

Reply:
I agree. Maybe we should do it in a dedicated separate PR though, what do you think?

Reply:
Dedicated PR is also fine.